src/main/resources/project/parameterForms/CreateSnapshot.xml

```diff
@@ -0,0 +1,79 @@
+<!--
+
+     Copyright 2015 Electric Cloud, Inc.
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+
+-->
+<editor>
+    
+    <formElement>
+        <type>entry</type>
+        <label>Application Name:</label>
+        <property>ApplicationName</property>
+        <documentation>Name of the application for the snapshot.</documentation>
+        <required>1</required>
+        <value>$[/myApplication/applicationName]</value>
+
+    </formElement>
+
+    <formElement>
+        <type>entry</type>
+        <label>Project Name:</label>
+        <property>ProjectName</property>
+        <documentation>Name of the project that contains the application.</documentation>
+        <required>1</required>
+    </formElement>
+
+    <formElement>
+        <type>entry</type>
+        <label>Snapshot Name:</label>
+        <property>SnapshotName</property>
+        <documentation>Name of the snapshot to be created.</documentation>
+        <required>1</required>
+    </formElement>
+
+    <formElement>
+        <type>entry</type>
+        <label>Environment Name:</label>
+        <property>EnvironmentName</property>
+        <documentation>For an environment snapshot, specify the name of the environment where the application is deployed.</documentation>
+        <required>0</required>
+        <value>$[/myEnvironment/environmentName]</value>
+    </formElement>
+
+    <formElement>
+        <type>entry</type>
+        <label>Environment Project Name:</label>
+        <property>EnvironmentProjectName</property>
+        <documentation>Name of the project that contains the environment, if different from the application project.</documentation>
```

Comment:
Add a default for the Environment project name also per Vinoo's comment:

 <value>$[/myEnvironment/projectName]</value>